### PR TITLE
Convert Workflow.Rnw to Workflow.Rmd

### PR DESCRIPTION
**Pull request by Oluwabukolab**
Outreachy applicant
@jwokaty @villafup @mcarlsn Goodday! Please kindly review my PR at your earliest convenience. Thank you

### CODE REVIEW

**Comment by jwokaty** [comment]
Thanks for your PR. While you wait for a review, I'll share the checklist we'll use for review, so that you can do your own review.

- [ ] the `.Rmd` file knits to `HTML`
- [ ] `R CMD build` runs without errors or timeouts
- [ ] the tarball from `R CMD build` contains the `HTML` (check with the
  following by substituting the package name and vignette name `tar ztf
  package_name.tar.gz | grep 'doc/vignette_name'`)
- [ ] the `.Rnw` file has been removed
- [ ] in the `DESCRIPTION`, `BiocStyle` and `knitr` are listed in `Suggests`
- [ ] in the `DESCRIPITON`, the line `VignetteBuilder: knitr` exists
- [ ] any added lines use the same spacing and indents as the existing
  document
- [ ] if agreed with the Maintainer, the contributor is in the author list
  in the `DESCRIPTION` file; refer to the contribution guide for acceptable
  formats
- [ ] if agreed with the Maintainer, the contributor is in the author list
  in the vignette's YAML; refer to the contribution guide for acceptable formats
- [ ] if this pull request includes a conversion from using the separate `Author`
  and `Maintainer` lines into the `Authors@R` vector, check that the
  `Authors@R` vector includes the maintainer as specified with `role='cre'`.
- [ ] if this pull request includes a conversion from using the separate `Author`
  and `Maintainer` lines into the `Authors@R` vector, the `Maintainer` line
  is completely removed.
- [ ] `HTML` document is representative of the `PDF` in content and
  in general the presentation
- [ ] Where the contributor was not able to preserve the content and presentation
  of the `PDF` is noted as a comment in the pull request
- [ ] the R Markdown file is representative of the Sweave document and follows
  best practices, such as replacing links to Bioconductor packages with calls
  to BiocStyle's `Biocpkg()`
  
I'm also adding the PDF as a reference: https://bioconductor.org/packages/3.17/bioc/vignettes/chipseq/inst/doc/Workflow.pdf

File: DESCRIPTION
Comment (jwokaty) [inline DESCRIPTION]: Please use the same indent as the rest of the file, which looks like 8 spaces/tab. Note that the second line of the person entry should also be indented more; for example

    Authors@R: c(
            person("Bioconductor Package Maintainer",
                    email = "maintainer@bioconductor.org", role = "cre"))

File: DESCRIPTION
Comment (jwokaty) [inline DESCRIPTION]: `knitr` has a lowercase k. (In R, capitalization is important.)

File: vignettes/Workflow.Rmd
Comment (jwokaty) [inline vignettes/Workflow.Rmd]: Remove extra blank line.

File: vignettes/Workflow.Rmd
Comment (jwokaty) [inline vignettes/Workflow.Rmd]: The PDF doesn't list any authors so it isn't clear who are the authors. Please remove them. It's okay to leave yourself as the converter.

File: vignettes/Workflow.Rmd
Comment (jwokaty) [inline vignettes/Workflow.Rmd]: The patterns following `[chipseq]{.sans-serif}` are packages, so I will ask you to replace them with code to generate canonical links: https://bioconductor.org/packages/release/bioc/vignettes/BiocStyle/inst/doc/AuthoringRmdVignettes.html#4_Style_macros. Packages can be from Bioconductor, CRAN, or from somewhere else so you might have to search. You can use https://bioconductor.org/packages/release/BiocViews.html#___Software to check if a package exists in Bioconductor.

Please also make changes for packages mentioned elsewhere in this document.

File: vignettes/Workflow.Rmd
Comment (jwokaty) [inline vignettes/Workflow.Rmd]: Use `message=FALSE` to hide loading messages rather than `results="hide"`. Also, be consistent about space after commas.

File: vignettes/Workflow.Rmd
Comment (jwokaty) [inline vignettes/Workflow.Rmd:51]: Add a blank line before line 55. We also have a blank line before and after code blocks. Please correct the spacing throughout the vignette.

File: vignettes/Workflow.Rmd
Comment (jwokaty) [inline vignettes/Workflow.Rmd]: Remove the extra blank lines: 92, 94.

File: vignettes/Workflow.Rmd
Comment (jwokaty) [inline vignettes/Workflow.Rmd]: Put a space between `#` and `Coverage`. Do this for other instances in this vignette.

**Comment by jwokaty** [comment]
Hi @Bukolab99, thanks for your PR. I've provided some initial feedback so you can work on it a little more. When you're finished, you can ask me to continue reviewing.

**Comment by Oluwabukolab** [comment]
> Hi @Bukolab99, thanks for your PR. I've provided some initial feedback so you can work on it a little more. When you're finished, you can ask me to continue reviewing.

I will work on the issues ASAP. Thank you

**Comment by Oluwabukolab** [comment]
@jwokaty HI! I have made the necessary changes you outlined. Please review Thank you!

**Comment by mcarlsn** [comment]
@Bukolab99 GM! As a FYI, I am in the process of reviewing your PR. Will post more comments later on. Thanks for your work on this.

**Comment by mcarlsn** [comment]
Great work on this conversion, @Bukolab99. I had one issue with a code chunk and an inline code edit, see comments and notes below, but everything else looked awesome. Let me know if you ran into the same code chunk error - once we figure that out and you made the inline code edit I suggested, this vignette will be in a good place for its next step. 

- [ ] the `.Rmd` file knits to `HTML` - the file knits great with the exception for one pesky section of code that begins with sumtab <- as.data.frame(rbind(total = xtabs(change, peakSummary)" error message is as follows: "Error in h(simpleError(msg, call)): error in evaluating the argument 'x' in selecting a method for function 'as.data.frame': error in evaluating the argument 'data' in selecting a method for function 'xtabs': 'subset' must be coercible to logical." Did you run into the same issue? @jwokaty flagging error - I'm doing some research to figure it out but haven't found a solution yet. This is an isolated chunk (i.e., no future syntax depends on it, so I was able to knit the script after adding "#" to this line), but still working on the error message.. 
- [x] `R CMD build` runs without errors or timeouts
- [x] the tarball from `R CMD build` contains the `HTML` (check with the
  following by substituting the package name and vignette name `tar ztf
  package_name.tar.gz | grep 'doc/vignette_name'`)
- [x] the `.Rnw` file has been removed
- [x] in the `DESCRIPTION`, `BiocStyle` and `knitr` are listed in `Suggests`
- [x] in the `DESCRIPITON`, the line `VignetteBuilder: knitr` exists
- [x] any added lines use the same spacing and indents as the existing
  document
- [x] if agreed with the Maintainer, the contributor is in the author list
  in the `DESCRIPTION` file; refer to the contribution guide for acceptable
  formats
- [x] if agreed with the Maintainer, the contributor is in the author list
  in the vignette's YAML; refer to the contribution guide for acceptable formats
- [x] if this pull request includes a conversion from using the separate `Author`
  and `Maintainer` lines into the `Authors@R` vector, check that the
  `Authors@R` vector includes the maintainer as specified with `role='cre'`.
- [x] if this pull request includes a conversion from using the separate `Author`
  and `Maintainer` lines into the `Authors@R` vector, the `Maintainer` line
  is completely removed.
- [x] `HTML` document is representative of the `PDF` in content and
  in general the presentation
- [x] Where the contributor was not able to preserve the content and presentation
  of the `PDF` is noted as a comment in the pull request
- [x] the R Markdown file is representative of the Sweave document and follows
  best practices, such as replacing links to Bioconductor packages with calls
  to BiocStyle's `Biocpkg()`

**Comment by Oluwabukolab** [comment]
> Great work on this conversion, @Bukolab99. I had one issue with a code chunk and an inline code edit, see comments and notes below, but everything else looked awesome. Let me know if you ran into the same code chunk error - once we figure that out and you made the inline code edit I suggested, this vignette will be in a good place for its next step.
> 
> * [ ]  the `.Rmd` file knits to `HTML` - the file knits great with the exception for one pesky section of code that begins with sumtab <- as.data.frame(rbind(total = xtabs(change, peakSummary)" error message is as follows: "Error in h(simpleError(msg, call)): error in evaluating the argument 'x' in selecting a method for function 'as.data.frame': error in evaluating the argument 'data' in selecting a method for function 'xtabs': 'subset' must be coercible to logical." Did you run into the same issue? @jwokaty flagging error - I'm doing some research to figure it out but haven't found a solution yet. This is an isolated chunk (i.e., no future syntax depends on it, so I was able to knit the script after adding "#" to this line), but still working on the error message..
> * [x]  `R CMD build` runs without errors or timeouts
> * [x]  the tarball from `R CMD build` contains the `HTML` (check with the
>   following by substituting the package name and vignette name `tar ztf package_name.tar.gz | grep 'doc/vignette_name'`)
> * [x]  the `.Rnw` file has been removed
> * [x]  in the `DESCRIPTION`, `BiocStyle` and `knitr` are listed in `Suggests`
> * [x]  in the `DESCRIPITON`, the line `VignetteBuilder: knitr` exists
> * [x]  any added lines use the same spacing and indents as the existing
>   document
> * [x]  if agreed with the Maintainer, the contributor is in the author list
>   in the `DESCRIPTION` file; refer to the contribution guide for acceptable
>   formats
> * [x]  if agreed with the Maintainer, the contributor is in the author list
>   in the vignette's YAML; refer to the contribution guide for acceptable formats
> * [x]  if this pull request includes a conversion from using the separate `Author`
>   and `Maintainer` lines into the `Authors@R` vector, check that the
>   `Authors@R` vector includes the maintainer as specified with `role='cre'`.
> * [x]  if this pull request includes a conversion from using the separate `Author`
>   and `Maintainer` lines into the `Authors@R` vector, the `Maintainer` line
>   is completely removed.
> * [x]  `HTML` document is representative of the `PDF` in content and
>   in general the presentation
> * [x]  Where the contributor was not able to preserve the content and presentation
>   of the `PDF` is noted as a comment in the pull request
> * [x]  the R Markdown file is representative of the Sweave document and follows
>   best practices, such as replacing links to Bioconductor packages with calls
>   to BiocStyle's `Biocpkg()`

Thank you @mcarlsn for commending my effort it really makes me happy. I will look into the issue popping up with the code chunk and an inline code edit and make the necessary modifications and hope the vignette will be in a good place for the next step. When I have solved the issue I will let you know. Have a lovely morning!

File: vignettes/Workflow.Rmd
Comment (villafup) [inline vignettes/Workflow.Rmd:427]: This loop is looking for the variable "_change_".  _change_ is populated in another loop on line 389.  

The variable is local to the loop. It, and the others inside that loop, don't exist outside the loop.

Comment (Oluwabukolab) [reply vignettes/Workflow.Rmd:427]: Hi @villafup Based on the error message  @mcarlsn outlined, it seems like the issue is with the subset function being used in combination with xtabs. The subset function is not being evaluated correctly, resulting in the error. Specifically, the error message mentions that the 'subset' argument must be coercible to logical, meaning it should be a valid logical expression or a logical vector.

To fix this issue, I want to review and correct the usage of the 'subset', 'as.data.frame', and 'xtabs' functions in the code, making sure that the 'subset' argument is a valid logical expression or vector. 

If you try running my converted code chunk you'd get the error messagge "Error in h(simpleError(msg, call)) : 
  error in evaluating the argument 'x' in selecting a method for function 'as.data.frame': object 'change' not found
Calls: <Anonymous> ... makeEnvForNames -> new.env -> environment -> " and if you try running the exact code chunk from the .Rwn sumtab <- 
    as.data.frame(rbind(total = xtabs(~ change, peakSummary),
                        promoter = xtabs(~ change, 
                          subset(peakSummary, inPromoter)),
                        upstream = xtabs(~ change, 
                          subset(peakSummary, inUpstream)),
                        gene = xtabs(~ change, subset(peakSummary, inGene))))
##cbind(sumtab, ratio = round(sumtab[, "down"] /  sumtab[, "up"], 3)) you'll get a different but similar error message "error in evaluating the argument 'x' in selecting a method for function 'as.data.frame': error in evaluating the argument 'data' in selecting a method for function 'xtabs': 'subset' must be coercible to logical
Calls: <Anonymous> ... subset -> .local -> evalqForSubset -> normSubsetIndex" . What do you advise please?

Comment (jwokaty) [reply vignettes/Workflow.Rmd:427]: Hi, sorry it took me awhile to look at this issue. I noticed that there are some formatting issues in the code blocks, which can be contributing to this issue. I didn't find exactly where the problem exists, but I tried copying over all code blocks from the .Rnw to the .Rmd and was able to knit and run `R CMD build`. 

From that, I made a patch (but renamed the extension from .patch to .txt so I could share it) so you can see the differences. If you've never looked at a patch or a diff, they show the differences between the files. I recommend looking it with something that has syntax highlighting.

[0001-Copy-code-blocks-from-.Rnw.txt](https://github.com/Bioconductor/chipseq/files/11160186/0001-Copy-code-blocks-from-.Rnw.txt)

You could also copy the code blocks from the .Rnw then run knit incrementally to identify the problem.  If you understand the content of the patch, you could also rename and it.

Comment (Oluwabukolab) [reply vignettes/Workflow.Rmd:427]: Hello @jwokaty Hope your day is going great? So I was able to knit and run R CMD build after copying over all code blocks from the .Rnw to the .Rmd like you advised. Now, I'm trying to compare both code from both files to find the cause of the issue in the .Rmd file. Hopeful I figure out soon and send a new commit. Thank you!

**Comment by Oluwabukolab** [comment]
Hello @mcarlsn I was able to resolve the issue with the code chunk and voila! It [knits](https://rpubs.com/Bukolab99/1024787) without no issue thanks to @jwokaty advice. Please review as to unable us move to the next step. Thank you.

File: DESCRIPTION
Comment (jwokaty) [inline DESCRIPTION]: We lost some packages that should be included in `Imports`: methods, stats, lattice, BiocGenerics, IRanges, GenomicRanges, ShortRead

File: DESCRIPTION
Comment (jwokaty) [inline DESCRIPTION]: Let's clarify and shorten the comment by rewriting `Converted vignette from Sweave to R Markdown / HTML.`

File: vignettes/Workflow.Rmd
Comment (jwokaty) [inline vignettes/Workflow.Rmd]: Let's remove `fig=TRUE` (this may be a setting only for Sweave) and replace `height` with `fig.height`. Please do this for the other plot code blocks.

**Comment by Oluwabukolab** [comment]
HI @jwokaty I have made the necessary changes you listed out please review. And I had issues committing and pushing because it kept committing the wrong files so I had to first delete all the files that wasn't meant to be on the repo before I had to improvise to commit and push the right files you made comment on.

File: vignettes/Workflow.Rmd
Comment (jwokaty) [inline vignettes/Workflow.Rmd]: Replace `converter` with `author`. This is currently preventing your name from appearing on the HTML.

File: vignettes/Workflow.Rmd
Comment (jwokaty) [inline vignettes/Workflow.Rmd]: Remove the space between the d and the comma in `%d ,`.

File: vignettes/Workflow.Rmd
Comment (jwokaty) [inline vignettes/Workflow.Rmd]: Use

    `r CRANpkg("lattice")`

for `lattice` since it is a CRAN package.

File: vignettes/Workflow.Rmd
Comment (jwokaty) [inline vignettes/Workflow.Rmd]: I noticed that in this vignette, a lot of the plots show up twice--not because of anything you did; I think I would have written the same code blocks as 199-203 and 205-207 because that's close to the .Rnw. Maybe this is the difference between how Sweave and R Markdown handle plots in code blocks or at least the `xyplot`, which I am not familiar with.

Before you do anything, I am going to ask another core team member's opinion since I usually don't deviate from the .Rnw

I want to propose removing the `plot(trellis.last.object())` blocks since they are just replotting the last object. For example, delete lines 205-207 and at line 200, add `fig.height=10` as an option in the opening part of the code block such that it becomes

   ```{r fig.height=10}

We don't include `echo=FALSE` because we do want the code to show up. We'd do this for any plots that appear twice.

File: DESCRIPTION
Comment (jwokaty) [inline DESCRIPTION]: Could you break the long lines on 17 and 18 and indent the second line, for example, as it had been on old line 9 and 10 for Imports?

**Review by jwokaty — CHANGES_REQUESTED** [review]
Hi @Bukolab99 I'm glad you hung in there and got it working. I'm sure it took effort and trial and error. 

I noticed you accidentally deleted the .gitignore file. Since it was there, could you add it back?

I noticed a few things to address before merging within the DESCRIPTION and .Rmd files.
I made a much longer note on line 207 after noticing plots were being duplicated--not because of anything you did--but because I think the R Markdown code blocks treat the plots differently than the Sweave format. I proposed removing the code that is duplicating the plots, but I am going to ask another team member's opinion since that would alter the original code. So please wait to make the changes addressing these duplicate plots. This may take time. You can make the other changes and if the delay is too long (let's say week?) for this part, we can consider this PR "approved" and I will take over the PR, make any adjustments, and merge it.

File: vignettes/Workflow.Rmd
Comment (Oluwabukolab) [reply vignettes/Workflow.Rmd]: Hi @jwokaty ,

Thank you for your feedback. I apologize for accidentally deleting the .gitignore file, and I have added it back to the repository.
I also appreciate your comments regarding the DESCRIPTION and .Rmd files. I have made other changes you commented on but I will hold off on making changes to address the duplicate plots as per your request and wait for the opinion of another team member, since it may require altering the original code. Please let me know if you need any further action from my end.
If the delay in resolving the duplicate plots issue takes longer than a week, we can consider this PR "approved" and I will be happy to hand it over to you for any necessary adjustments and merging.

Thank you for your guidance.

Comment (jwokaty) [reply vignettes/Workflow.Rmd]: I think we can proceed with removing the duplicates. There is one special one that should be modified differently and I will mark it on the .Rmd.

File: vignettes/Workflow.Rmd
Comment (jwokaty) [inline vignettes/Workflow.Rmd]: Lines 233-265 are the code blocks we'll do differently because we have the comments on line 260 that should stay with the code block. (If you look at them in the knitted file they show up after the plot rather than with the rest of the code.) So we should save the plot to a variable and later 'return' that variable in another code block. Please edit as follows:

On Line 248, set the `xyplot` to a `plt`:

    plt <- xyplot(log(count) ~ depth | sample, as.data.frame(depth.islands),

On line 264, replace `plot(trellis.last.object())` with `plt`.

File: vignettes/Workflow.Rmd
Comment (jwokaty) [inline vignettes/Workflow.Rmd]: Add `eval=FALSE` as I believe this plot isn't displayed in the original PDF.

**Comment by Oluwabukolab** [comment]
Good morning @jwokaty I have made the changes! Please review. Thank you

**Comment by jwokaty** [comment]
@Bukolab99 Thanks for your contribution!

**Comment by Oluwabukolab** [comment]
> @Bukolab99 Thanks for your contribution!

Thank you for your guidance @jwokaty.

**Pull request merged**
